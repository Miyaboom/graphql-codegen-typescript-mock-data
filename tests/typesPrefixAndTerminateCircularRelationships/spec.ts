import { plugin } from '../../src';
import testSchema from './schema';

it('should support typesPrefix and terminateCircularRelationships at the same time', async () => {
    const result = await plugin(testSchema, [], {
        prefix: 'mock',
        typesPrefix: 'Mock',
        terminateCircularRelationships: true,
    });

    expect(result).toBeDefined();
    expect(result).toContain(
        "a: overrides && overrides.hasOwnProperty('a') ? overrides.a! : relationshipsToOmit.has('A') ? {} as MockA : mockA({}, relationshipsToOmit)",
    );
    expect(result).toMatchSnapshot();
});
