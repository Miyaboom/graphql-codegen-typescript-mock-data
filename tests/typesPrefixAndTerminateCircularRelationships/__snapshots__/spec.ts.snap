// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should support typesPrefix and terminateCircularRelationships at the same time 1`] = `
"
export const mockA = (overrides?: Partial<MockA>, _relationshipsToOmit: Set<string> = new Set()): MockA => {
    const relationshipsToOmit: Set<string> = new Set(_relationshipsToOmit);
    relationshipsToOmit.add('A');
    return {
        b: overrides && overrides.hasOwnProperty('b') ? overrides.b! : relationshipsToOmit.has('B') ? {} as MockB : mockB({}, relationshipsToOmit),
    };
};

export const mockB = (overrides?: Partial<MockB>, _relationshipsToOmit: Set<string> = new Set()): MockB => {
    const relationshipsToOmit: Set<string> = new Set(_relationshipsToOmit);
    relationshipsToOmit.add('B');
    return {
        a: overrides && overrides.hasOwnProperty('a') ? overrides.a! : relationshipsToOmit.has('A') ? {} as MockA : mockA({}, relationshipsToOmit),
    };
};
"
`;
